torchelastic/timer/local_timer.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+import os
+import signal
+import time
+from queue import Empty
+from typing import Dict, List
+
+from .api import RequestQueue, Timer, TimerClient, TimerRequest, TimerServer
+
+
+class LocalTimerClient(TimerClient):
+    def __init__(self, mp_queue):
+        self._mp_queue = mp_queue
+
+    def acquire(self, scope_id, expiration_time) -> Timer:
```

Comment:
You could just do

```py
@contextlib.contextmanager
def acquire(...):
        pid = os.getpid()
        acquire_request = TimerRequest(pid, scope_id, expiration_time)
        self._mp_queue.put(acquire_request) 
        yield
        release_request = TimerRequest(pid, scope_id, -1)
        self._mp_queue.put(release_request) 
```

and get rid of the Timer class and release entirely

Reply:
hmmm this would make the interface a bit more ambiguous to implement no? basically all implementations need to know that the `acquire` method should be a context manager.

Reply:
actually, after giving this a thought I think I have a clean way to get rid of the `Timer` class. What I did was to annotate the `expires()` method with `@contextmanager` and made `TimerClient#acquire()` and `TimerClient#release()` methods voids:

```
@contextmanager
expires(after, scope_id, client):
   #... omitted...
   client.acquire(scope_id, time.time() + after
   try:
      yield
   finally:
       client.release(scope_id)
```